viz_scripts/survey_responses.ipynb

```diff
@@ -0,0 +1,403 @@
+    "def generate_debug_df(program, include_test_users, full_df, labeled_df):\n",
+    "    debug_df = pd.DataFrame.from_dict({\n",
+    "            \"year\": year,\n",
+    "            \"month\": month,\n",
+    "            \"Registered_participants\": len(scaffolding.get_participant_uuids(program, include_test_users)),\n",
+    "            \"Participants_with_at_least_one_trip\": scaffolding.unique_users(full_df),\n",
+    "            \"Participant_with_at_least_one_labeled_trip\": scaffolding.unique_users(labeled_df),\n",
+    "            \"Trips_with_at_least_one_label\": len(labeled_df)\n",
+    "            },\n",
+    "        orient='index', columns=[\"value\"])\n",
+    "    \n",
+    "    return debug_df\n",
+    "\n",
+    "#color dictionary\n",
+    "def mapping_color_surveys(dic_options):\n",
+    "    dictionary_values = (list(collections.OrderedDict.fromkeys(dic_options.values())))\n",
+    "    \n",
+    "    colors = {}\n",
+    "    for i in range(len(dictionary_values)):\n",
+    "        colors[dictionary_values[i]] = plt.cm.tab10.colors[i%10]\n",
+    "    \n",
+    "    return colors\n",
+    "\n",
+    "#workaround missing colors\n",
+    "def get_survey_colors(labels, existing_map):\n",
+    "    color_map = {}\n",
+    "    for label in labels:\n",
+    "        l_labels = label.split(\"\\n\")\n",
+    "        color = (0,0,0)\n",
+    "        n = 0\n",
+    "        for i in range(len(l_labels)):\n",
+    "            try:\n",
+    "                color = tuple(map(lambda i, j: i + j, color, existing_map[l_labels[i]]))\n",
+    "            except:\n",
+    "                print(\"missing color\")\n",
+    "                color = tuple(map(lambda i, j: i + j, color, (0.1, 0.2, 0.5)))\n",
+    "            \n",
+    "            n += 1\n",
+    "            \n",
+    "        print(color, n)\n",
+    "        color = [x/n for x in color]\n",
+    "        \n",
+    "        color_map[label] = color\n",
+    "    \n",
+    "    color_map['Other'] = existing_map['Other']\n",
+    "    \n",
+    "    return color_map"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5319980e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#list of all surveys that are not a \"UserProfileSurvey\"\n",
+    "survey_list = list(survey_info['surveys'].keys())\n",
+    "survey_list.remove('UserProfileSurvey')\n",
+    "sheet_list = get_sheet_links(survey_list)\n",
+    "\n",
+    "print('survey sheets: ', sheet_list)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3b355efd",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tq = scaffolding.get_time_query(year, month)\n",
+    "all_confirmed_trips = scaffolding.load_all_confirmed_trips(tq)\n",
+    "#we need to filter out trips (based on if including test users)\n",
+    "all_confirmed_trips = scaffolding.filter_composite_trips(all_confirmed_trips, program, include_test_users)\n",
+    "\n",
```

Comment:
I think this should all go into scaffolding (similar to `load_viz_notebook_sensor_inference_data`)